tests/integration/test_cogs_rates.py

```diff
@@ -20,6 +22,28 @@ def rates_cog(sample_bot):
             pass
 
 
+@pytest.mark.asyncio
+async def test_send_embed(rates_cog):
+    # Define an embed that matches what should be published by the cog
+    sample_embed = discord.Embed(
+        description="test",
+        title="ARK's Official server rates have just been updated!",
+        color=0x63BCC3,
+    )
+    sample_embed.set_image(url=EMBED_IMAGE)
+
+    # Publish the embed using the Rates cog
+    await rates_cog.send_embed(
+        description="test", url="www.mysite.com/dynamicconfig.ini"
+    )
+
+    # Ensure embed appears in the queue and nothing else.
+    assert dpytest.verify().message().embed(embed=sample_embed)
+    assert dpytest.verify().message().nothing()
```

Comment:
Basically it seems as though an embed gets sent into a message queue and then can be consumed by the `dpytest.verify()` methods. [Here](https://github.com/CraftSpider/dpytest/blob/2b16e22325d19a75b6b9b83c95ce754be65af3ea/tests/test_verify_embed.py#L6-L18) is the reference I found to build this from.

The 1st call confirms that an embed is on the message queue, which is then consumed. The 2nd call confirms the queue is now empty.

Reply:
Man thats pretty neat!